lib/buildlog/mod.rs

```diff
@@ -127,3 +187,7 @@ fn parse_hex(s: &RawStr) -> Option<u64> {
 		.ok()
 		.and_then(|s| u64::from_str_radix(s, 16).ok())
 }
+
+fn as_millis(d: Duration) -> u32 {
+	d.as_secs() as u32 * 1000 + d.subsec_millis()
```

Comment:
How about overflows? the `as u32` part just silenty truncates, but the `* 1000` and `+` part panic when they overflow. You probably at least want consistency here.